### ISSUE TITLE
Hybrid Search Bug With Long Lists[BUG]

### ISSUE THREAD

**Issue by avihe** [milvus-io/pymilvus#690]
**Describe the bug**
Hybrid search does not perform a full search when giving a long list in the query for example expr="count in [1,2,3,....,n]".

**Steps/Code to reproduce behavior**
1. Create a Milvus collection
2. search_params = {"metric_type": "L2", "params": {"nprobe": 1}}
3. filter_list = collection.search([vec], 'imVec', search_params, 1000)[0].ids
4. ans = collection.search([vec], 'imVec', search_params, 1000, expr=f"count in {filter_list}")[0].ids
ans != filter_list    ----->>>> BUG

**Expected behavior**
ans == filter_list

**Environment details**
Linux, 64 GB RAM, 8 CPUs
Docker-compose (milvus cluster)
Milvus v2.0.0rc5

**Comment by XuanYang-cn** [comment]
@avihe What index did you create?

**Comment by avihe** [comment]
Ivf_pq

**Comment by XuanYang-cn** [comment]
> Ivf_pq

@avihe 1. What's the index param when creating IVF PQ index?

2. How many numofRows vectors are in  the collection?

**Comment by avihe** [comment]
Hi, 
I have 5M rows.
index_params = {'index_type': 'IVF_PQ',
         'metric_type': 'L2',
         'params': {
             'nlist': 100,
             "m": 64
         }}
Thanks!

**Comment by XuanYang-cn** [comment]
```python
"""
https://github.com/milvus-io/pymilvus/issues/690
1. create, insert, create_index, load
2. search without expr, get a list of ids
3. search with expr f"film_id in {list of ids}"
"""

from pymilvus import Collection, FieldSchema, CollectionSchema, DataType
from data import gen_float_vectors

from functools import wraps
from time import time
import multiprocessing
import os


COLLECTION_NAME = "_690"
NUM = 5_000
DIM = 128


def timeit(f):
    @wraps(f)
    def wrap(*args, **kw):
        ts = time()
        result = f(*args, **kw)
        te = time()

        print(f"func:{f.__name__} - [{args}] took: {te-ts:8.4f}s")
        return result
    return wrap


class MultiProcessing:
    def __init__(self):
        from pymilvus import connections
        m = connections.connect()
        if m.has_collection(COLLECTION_NAME):
            m.drop_collection(COLLECTION_NAME)

    @staticmethod
    @timeit
    def task(j):
        from pymilvus import connections
        schema = CollectionSchema(
            fields=[
                FieldSchema("film_id", DataType.INT64, is_primary=True, auto_id=True),
                FieldSchema("films", dtype=DataType.FLOAT_VECTOR, dim=128)
            ],
        )
        connections.connect()
        films = Collection(COLLECTION_NAME, schema=schema)

        for i in range(100):
            print(f"The {os.getpid()} {i:3}th - total num {NUM}, generate entities")
            ve = gen_float_vectors(NUM, DIM)
            films.insert([ve])
            print(f"The {os.getpid()} {i:3}th - insert")

    def insert(self):
        with multiprocessing.Pool(10) as pool:
            pool.map(self.task, [1] * 10)

    def run(self):
        # Multiprocessing insert 5M with 100 * 10 * 5000
        self.insert()

    @staticmethod
    def op():
        from pymilvus import connections
        schema = CollectionSchema(
            fields=[
                FieldSchema("film_id", DataType.INT64, is_primary=True, auto_id=True),
                FieldSchema("films", dtype=DataType.FLOAT_VECTOR, dim=128)
            ],
        )
        connections.connect()
        films = Collection(COLLECTION_NAME, schema=schema)

        index_params = {
            "index_type": "IVF_PQ",
            "metric_type": "L2",
            "params": {"nlist": 100, "m": 64}
        }

        films.create_index("films", index_params)
        print(f"index films")
        print(f"Total num {films.num_entities}")

        films.load()
        print(f"load films")

        toSearch = gen_float_vectors(1, DIM)
        search_param = {
            "data": toSearch,
            "anns_field": "films",
            "param": {"metric_type": "L2", "params": {"nprobe": 1}},
            "limit": 10,
        }

        filter_lists = films.search(**search_param)[0].ids
        print(f"search 1")
        ans = films.search(**search_param, expr=f"film_id in {filter_lists}")[0].ids

        print(f"search 2")
        print(f"Expected both true")
        print(len(ans) == len(filter_lists))

        print(ans == filter_lists)


if __name__ == "__main__":
    multiprocessing.set_start_method('spawn')
    mp = MultiProcessing()
    mp.run()

    mp.op()
```

I use this script, but the result is true, can you tell me what's the difference of your code with mine?

**Comment by XuanYang-cn** [comment]
is `count` in `f"count in {filter_list}")[0].ids` your id field name?

**Comment by avihe** [comment]
yes

**Comment by XuanYang-cn** [comment]
@avihe What's the collection schema?

**Comment by fishpenguin** [comment]
The ans and filter_lists has the same content but with different sort, so the assert is failed.
Search result does not promise the sort.

**Comment by avihe** [comment]
I'll try to recreate a simple code that regenerates the problem and update

**Comment by avihe** [comment]
Here is the code that reproduces my problem:


from pymilvus import connections, FieldSchema, CollectionSchema, DataType, Collection
import numpy as np

DIM = 128
host = 'localhost'
port = 19530
connections.connect(host=host, port=port)
db_name = 'debug3'

# define db
default_fields = [
    FieldSchema(name="count", dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name="imVec", dtype=DataType.FLOAT_VECTOR, dim=DIM),
]
default_schema = CollectionSchema(fields=default_fields, description=db_name)
collection = Collection(name=db_name, schema=default_schema)

# insert vectors
np.random.seed(123)
vectors = np.random.random_integers(0, 10, int(1e4)*128).reshape(-1, DIM).astype(np.float32).tolist()
mr = collection.insert([vectors])

#index
index_params = {'index_type': 'IVF_PQ',
                'metric_type': 'L2',
                'params': {
                    'nlist': 100,
                    "m": 32
                }}

collection.create_index(field_name='imVec', index_params=index_params)

# query
collection.load()
query_vector = [0] * DIM
k = 100
search_params = {"metric_type": 'L2', "params": {"nprobe": 1}}
ans = collection.search([query_vector], 'imVec', search_params, k)[0]
print(f"origin ans.ids : {ans.ids}")
ans2 = collection.search([query_vector], 'imVec', search_params, k, expr=f"count in {ans.ids}")[0]
print(f"expression search ans2.ids : {ans2.ids}")






This is my output:
origin ans.ids : [427722607888569117, 427722607888564887, 427722607888568448, 427722607888571400, 427722607888570597, 427722607888564895, 427722607888570694, 427722607888568885, 427722607888572865, 427722607888571700, 427722607888568602, 427722607888572135, 427722607888573141, 427722607888569724, 427722607888564828, 427722607888571381, 427722607888570816, 427722607888565357, 427722607888568889, 427722607888569044, 427722607888572332, 427722607888563322, 427722607888564537, 427722607888570767, 427722607888572159, 427722607888566491, 427722607888566185, 427722607888570158, 427722607888572882, 427722607888569528, 427722607888567095, 427722607888565443, 427722607888572815, 427722607888563519, 427722607888570709, 427722607888564928, 427722607888568946, 427722607888570483, 427722607888564618, 427722607888571558, 427722607888570559, 427722607888568547, 427722607888572351, 427722607888565820, 427722607888568086, 427722607888564888, 427722607888572766, 427722607888572720, 427722607888566911, 427722607888567501, 427722607888568175, 427722607888566804, 427722607888571532, 427722607888566199, 427722607888569785, 427722607888568361, 427722607888568140, 427722607888566806, 427722607888571868, 427722607888572505, 427722607888567549, 427722607888567827, 427722607888566764, 427722607888564294, 427722607888564554, 427722607888565156, 427722607888570646, 427722607888564912, 427722607888568923, 427722607888569075, 427722607888571901, 427722607888572301, 427722607888572893, 427722607888568827, 427722607888570090, 427722607888567845, 427722607888569365, 427722607888570163, 427722607888564588, 427722607888572040, 427722607888571494, 427722607888572639, 427722607888568676, 427722607888569574, 427722607888563628, 427722607888569858, 427722607888563828, 427722607888563894, 427722607888571874, 427722607888571256, 427722607888564597, 427722607888571334, 427722607888567202, 427722607888569350, 427722607888572404, 427722607888569983, 427722607888564205, 427722607888567074, 427722607888567859, 427722607888566897]
expression search ans2.ids : [427722607888569858, 427722607888571874, 427722607888572404]

**Comment by fishpenguin** [comment]
This is a [bug](https://github.com/milvus-io/milvus/issues/7362) of milvus-2.0.0-RC5 and has been fixed in milvus-2.0.0-RC6.
Please update your milvus to RC6.

**Comment by avihe** [comment]
Great, thanks!! Closing this issue
